src/WebUI/ImageDetails/ImageDetails.tsx

```diff
@@ -48,6 +48,7 @@ export class ImageDetails extends BaseComponent<IImageDetailsProperties, IImageD
     private _getMainHeading(): JSX.Element | null {
         const imageDetails = this.props.imageDetails;
         this._displayImageName = Utils.extractDisplayImageName(imageDetails.imageName);
+        const jobName: string = format("{0} #{1} on {2}", imageDetails.jobName || "", imageDetails.pipelineVersion || "", imageDetails.pipelineName || "");
```

Comment:
please note if we do not have any of these details, we should not show "# on " string, we should skip the subtext altogether.

Reply:
Added changes to extract url from labels and using that in RunDetails